<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>UI</class>
 <widget class="QMainWindow" name="UI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1153</width>
    <height>911</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>查找可视化</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <widget class="QScrollArea" name="scrollArea">
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>1129</width>
         <height>836</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QTabWidget" name="tabWidget">
          <property name="currentIndex">
           <number>1</number>
          </property>
          <widget class="QWidget" name="tab">
           <attribute name="title">
            <string>列表查找</string>
           </attribute>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <widget class="QWidget" name="widget" native="true">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>5555</width>
                  <height>55555</height>
                 </size>
                </property>
                <property name="styleSheet">
                 <string notr="true"/>
                </property>
                <layout class="QHBoxLayout" name="horizontalLayout_9">
                 <item>
                  <widget class="QWidget" name="widget_3" native="true">
                   <layout class="QHBoxLayout" name="horizontalLayout_10">
                    <property name="spacing">
                     <number>0</number>
                    </property>
                    <property name="leftMargin">
                     <number>0</number>
                    </property>
                    <property name="topMargin">
                     <number>0</number>
                    </property>
                    <property name="rightMargin">
                     <number>0</number>
                    </property>
                    <property name="bottomMargin">
                     <number>0</number>
                    </property>
                    <item>
                     <layout class="QVBoxLayout" name="verticalLayout_7"/>
                    </item>
                   </layout>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="Line" name="line_7">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout">
                <item>
                 <widget class="QWidget" name="widget_2" native="true">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="minimumSize">
                   <size>
                    <width>0</width>
                    <height>100</height>
                   </size>
                  </property>
                  <layout class="QVBoxLayout" name="verticalLayout">
                   <item>
                    <layout class="QGridLayout" name="gridLayout">
                     <item row="0" column="0">
                      <widget class="QLabel" name="label">
                       <property name="text">
                        <string>快速初始化：</string>
                       </property>
                      </widget>
                     </item>
                     <item row="0" column="1">
                      <widget class="QLineEdit" name="ListInital"/>
                     </item>
                     <item row="1" column="0">
                      <widget class="QLabel" name="label_2">
                       <property name="text">
                        <string>查找目标：</string>
                       </property>
                      </widget>
                     </item>
                     <item row="1" column="1">
                      <widget class="QLineEdit" name="ListTarget"/>
                     </item>
                     <item row="2" column="0">
                      <widget class="QLabel" name="label_3">
                       <property name="text">
                        <string>查找方法：</string>
                       </property>
                      </widget>
                     </item>
                     <item row="2" column="1">
                      <widget class="QComboBox" name="SearchWay"/>
                     </item>
                    </layout>
                   </item>
                   <item>
                    <widget class="QPushButton" name="start">
                     <property name="text">
                      <string>开始</string>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item>
                 <widget class="Line" name="line_3">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QWidget" name="widget_6" native="true">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="minimumSize">
                   <size>
                    <width>500</width>
                    <height>200</height>
                   </size>
                  </property>
                  <property name="maximumSize">
                   <size>
                    <width>500</width>
                    <height>200</height>
                   </size>
                  </property>
                  <layout class="QHBoxLayout" name="horizontalLayout_8">
                   <item>
                    <layout class="QVBoxLayout" name="verticalLayout_6"/>
                   </item>
                  </layout>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_2">
           <attribute name="title">
            <string>哈希查找</string>
           </attribute>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <widget class="QWidget" name="HashWidget" native="true">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
               </widget>
              </item>
              <item>
               <widget class="Line" name="line">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QWidget" name="widget_8" native="true">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>350</height>
                 </size>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_5">
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <property name="bottomMargin">
                  <number>0</number>
                 </property>
                 <item>
                  <widget class="QTabWidget" name="tabWidget_2">
                   <property name="currentIndex">
                    <number>1</number>
                   </property>
                   <widget class="QWidget" name="tab_3">
                    <property name="sizePolicy">
                     <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                      <horstretch>0</horstretch>
                      <verstretch>0</verstretch>
                     </sizepolicy>
                    </property>
                    <attribute name="title">
                     <string>操控界面</string>
                    </attribute>
                    <layout class="QHBoxLayout" name="horizontalLayout_11">
                     <item>
                      <widget class="QWidget" name="widget_7" native="true">
                       <layout class="QHBoxLayout" name="horizontalLayout_12">
                        <property name="spacing">
                         <number>0</number>
                        </property>
                        <property name="leftMargin">
                         <number>0</number>
                        </property>
                        <property name="topMargin">
                         <number>0</number>
                        </property>
                        <property name="rightMargin">
                         <number>0</number>
                        </property>
                        <property name="bottomMargin">
                         <number>0</number>
                        </property>
                        <item>
                         <layout class="QHBoxLayout" name="horizontalLayout_4">
                          <property name="spacing">
                           <number>0</number>
                          </property>
                          <item>
                           <widget class="Line" name="line_4">
                            <property name="orientation">
                             <enum>Qt::Vertical</enum>
                            </property>
                           </widget>
                          </item>
                          <item>
                           <widget class="QWidget" name="widget_4" native="true">
                            <property name="sizePolicy">
                             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                              <horstretch>0</horstretch>
                              <verstretch>0</verstretch>
                             </sizepolicy>
                            </property>
                            <property name="minimumSize">
                             <size>
                              <width>0</width>
                              <height>0</height>
                             </size>
                            </property>
                            <layout class="QHBoxLayout" name="horizontalLayout_14">
                             <property name="spacing">
                              <number>0</number>
                             </property>
                             <property name="leftMargin">
                              <number>0</number>
                             </property>
                             <property name="topMargin">
                              <number>0</number>
                             </property>
                             <property name="rightMargin">
                              <number>0</number>
                             </property>
                             <property name="bottomMargin">
                              <number>0</number>
                             </property>
                             <item>
                              <layout class="QVBoxLayout" name="verticalLayout_4">
                               <property name="spacing">
                                <number>0</number>
                               </property>
                               <item>
                                <widget class="Line" name="line_2">
                                 <property name="orientation">
                                  <enum>Qt::Horizontal</enum>
                                 </property>
                                </widget>
                               </item>
                               <item>
                                <layout class="QGridLayout" name="gridLayout_2">
                                 <property name="spacing">
                                  <number>0</number>
                                 </property>
                                 <item row="0" column="0">
                                  <widget class="QLabel" name="label_4">
                                   <property name="text">
                                    <string>哈希数组大小：</string>
                                   </property>
                                  </widget>
                                 </item>
                                 <item row="0" column="1">
                                  <widget class="QLineEdit" name="HashSize"/>
                                 </item>
                                 <item row="1" column="0">
                                  <widget class="QLabel" name="label_5">
                                   <property name="text">
                                    <string>Hash函数：</string>
                                   </property>
                                  </widget>
                                 </item>
                                 <item row="1" column="1">
                                  <widget class="QComboBox" name="HashFuntion"/>
                                 </item>
                                 <item row="2" column="0">
                                  <widget class="QLabel" name="label_6">
                                   <property name="text">
                                    <string>碰撞处理方法：</string>
                                   </property>
                                  </widget>
                                 </item>
                                 <item row="2" column="1">
                                  <widget class="QComboBox" name="Collapes"/>
                                 </item>
                                 <item row="3" column="0">
                                  <widget class="QLabel" name="label_7">
                                   <property name="text">
                                    <string>插入目标Value：</string>
                                   </property>
                                  </widget>
                                 </item>
                                 <item row="3" column="1">
                                  <widget class="QLineEdit" name="KeyValue"/>
                                 </item>
                                </layout>
                               </item>
                               <item>
                                <widget class="QPushButton" name="MapInsert">
                                 <property name="text">
                                  <string>插入</string>
                                 </property>
                                </widget>
                               </item>
                               <item>
                                <layout class="QHBoxLayout" name="horizontalLayout_13">
                                 <item>
                                  <widget class="QLabel" name="label_8">
                                   <property name="text">
                                    <string>查找目标Key:   </string>
                                   </property>
                                  </widget>
                                 </item>
                                 <item>
                                  <widget class="QLineEdit" name="KeyEditor"/>
                                 </item>
                                </layout>
                               </item>
                               <item>
                                <widget class="QPushButton" name="MapSearch">
                                 <property name="text">
                                  <string>查找</string>
                                 </property>
                                </widget>
                               </item>
                               <item>
                                <widget class="Line" name="line_8">
                                 <property name="orientation">
                                  <enum>Qt::Horizontal</enum>
                                 </property>
                                </widget>
                               </item>
                              </layout>
                             </item>
                            </layout>
                           </widget>
                          </item>
                          <item>
                           <widget class="Line" name="line_5">
                            <property name="orientation">
                             <enum>Qt::Vertical</enum>
                            </property>
                           </widget>
                          </item>
                          <item>
                           <widget class="QWidget" name="widget_5" native="true">
                            <property name="sizePolicy">
                             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                              <horstretch>0</horstretch>
                              <verstretch>0</verstretch>
                             </sizepolicy>
                            </property>
                            <property name="minimumSize">
                             <size>
                              <width>500</width>
                              <height>0</height>
                             </size>
                            </property>
                            <property name="maximumSize">
                             <size>
                              <width>16777215</width>
                              <height>111111</height>
                             </size>
                            </property>
                            <layout class="QHBoxLayout" name="horizontalLayout_7">
                             <property name="spacing">
                              <number>0</number>
                             </property>
                             <property name="leftMargin">
                              <number>0</number>
                             </property>
                             <property name="topMargin">
                              <number>0</number>
                             </property>
                             <property name="rightMargin">
                              <number>0</number>
                             </property>
                             <property name="bottomMargin">
                              <number>0</number>
                             </property>
                             <item>
                              <widget class="QTextBrowser" name="textBrowser"/>
                             </item>
                            </layout>
                           </widget>
                          </item>
                         </layout>
                        </item>
                       </layout>
                      </widget>
                     </item>
                    </layout>
                   </widget>
                   <widget class="QWidget" name="tab_4">
                    <attribute name="title">
                     <string>设置界面</string>
                    </attribute>
                    <layout class="QVBoxLayout" name="verticalLayout_8">
                     <item>
                      <widget class="QLabel" name="label_9">
                       <property name="frameShape">
                        <enum>QFrame::NoFrame</enum>
                       </property>
                       <property name="lineWidth">
                        <number>2</number>
                       </property>
                       <property name="midLineWidth">
                        <number>3</number>
                       </property>
                       <property name="text">
                        <string>自定义哈希函数</string>
                       </property>
                       <property name="textFormat">
                        <enum>Qt::AutoText</enum>
                       </property>
                       <property name="alignment">
                        <set>Qt::AlignCenter</set>
                       </property>
                       <property name="margin">
                        <number>5</number>
                       </property>
                       <property name="textInteractionFlags">
                        <set>Qt::LinksAccessibleByMouse|Qt::TextEditable</set>
                       </property>
                      </widget>
                     </item>
                     <item>
                      <widget class="QLineEdit" name="FunName">
                       <property name="alignment">
                        <set>Qt::AlignCenter</set>
                       </property>
                       <property name="placeholderText">
                        <string>请输入函数名称</string>
                       </property>
                      </widget>
                     </item>
                     <item>
                      <widget class="QPlainTextEdit" name="FunctionPlainTextEdit">
                       <property name="frameShape">
                        <enum>QFrame::Box</enum>
                       </property>
                       <property name="frameShadow">
                        <enum>QFrame::Plain</enum>
                       </property>
                       <property name="lineWidth">
                        <number>2</number>
                       </property>
                       <property name="midLineWidth">
                        <number>0</number>
                       </property>
                      </widget>
                     </item>
                     <item>
                      <layout class="QHBoxLayout" name="horizontalLayout_15">
                       <item>
                        <widget class="QPushButton" name="AddHashFunction">
                         <property name="text">
                          <string>添加</string>
                         </property>
                        </widget>
                       </item>
                       <item>
                        <widget class="QPushButton" name="ClearHashFunction">
                         <property name="text">
                          <string>清空</string>
                         </property>
                        </widget>
                       </item>
                      </layout>
                     </item>
                    </layout>
                   </widget>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="Line" name="line_6">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1153</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
